src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -293,13 +293,13 @@ ExpressionSyntax initializer
 			//   int[] foo = { 1, 2, 3 };
 			var typeToCheck = typeInfo.Type ?? typeInfo.ConvertedType;
 
-			if ( initializer is ObjectCreationExpressionSyntax objCreation ) {
+			if ( initializer is BaseObjectCreationExpressionSyntax _ ) {
 				// When we have a new T() we don't need to worry about the value
 				// being anything other than an instance of T.
 				return (
 					typeToCheck,
 					ImmutableTypeKind.Instance,
-					() => objCreation.Type.GetLocation()
+					() => initializer.GetLocation()

```

Comment:
Always throwing the diagnostic on the initializer as a whole seemed more consistent to me anyways.
(`BaseObjectCreationExpressionSyntax` doesn't have `Type`, obviously, because `ImplicitObjectCreationExpressionSyntax` doesn't)